File: 19:08:2019/Answer.py
import copy

class List(object):
	class _ListNode(object):
		def __init__(self, x):
			self.val = x
			self.next = None

	def __init__(self, x):
		self.length = 1
		self.root = List._ListNode(x)
		self.nextNode = self.root

	def addRoot(self, x):
		self.length += 1
		temp = self.root
		self.root = List._ListNode(x)
		self.root.next = temp

	def add(self, x):
		self.length += 1
		temp = self.root
		
		while temp.next:
			temp = temp.next
		
		temp.next = List._ListNode(x)
	
	def toString(self):
		string = ""
		self.reset()
		while self.nextNode:
			string += str(self.nextNode.val)
			self.moveToNext()
		self.reset()
		string = string[::-1]
		return string

	def moveToNext(self):
		self.nextNode = self.nextNode.next
		return self.nextNode

	def reset(self):
		self.nextNode = self.root

class Solution:
	def addTwoNumbers(self, l1, l2):
		result = None
		if l2.length > l1.length:
			(l1, l2) = (l2, l1)
		carry = False
		l1.reset()
		l2.reset()
		iterator = l2.nextNode
		while iterator:
			x = iterator.val + l1.nextNode.val
			if carry:
				x += 1
			if x > 9:
				x %= 10
				carry = True
			else:
				carry = False
			if result == None:
				result = List(x)
			else:
				result.add(x)
			iterator = l2.moveToNext()
			l1.moveToNext()

		if l2.length != l1.length:
			i = 0
			while i < (l1.length - l2.length):
				result.add(l1.nextNode.val)
				result.moveToNext()
				l1.moveToNext()
				i += 1
		
		l1.reset()
		l2.reset()
		result.reset()
		return result
		
print ("Please enter the first number: ")
inputString = input()
l1 = None
for c in inputString:
	if l1 == None:
		l1 = List(int(c))
	else:
		l1.addRoot(int(c))

print ("Please enter the second number: ")
inputString = input()
l2 = None
for c in inputString:
	if l2 == None:
		l2 = List(int(c))
	else:
		l2.addRoot(int(c))


result = Solution().addTwoNumbers(l1, l2)
print (result.toString())

File: 19:08:2019/Answer.java
import java.lang.*;
import java.io.*;
import java.util.*;

class List {
    class ListNode {
        private int val;
        private List.ListNode next;

        public ListNode(int x) {
            this.val = x;
            this.next = null;
        }

        int getVal() {
            return this.val;
        }

        void setVal(int x) {
            this.val = x;
        }

        List.ListNode getNext() {
            return this.next;
        }

        void setNext(List.ListNode next) {
            this.next = next;
        }
    }

    private int length = 0;
    private List.ListNode root;
    private List.ListNode iterator;

    public List(int x) {
        root = new List.ListNode(x);
        iterator = root;
    }

    public void addRoot(int x) {
        length++;
        List.ListNode temp = root;
        root = new List.ListNode(x);
        root.setNext(temp);
    }

    public void add(int x) {
        length++;
        List.ListNode temp = root;

        while(temp.getNext() != null) {
            temp = temp.getNext();
        }

        temp.setNext(new List.ListNode(x));
    }

    public String toString() {
        String str = "";
        this.reset();

        while (this.iterator != null) {
            str += this.iterator.getVal();
            this.moveToNext();
        }

        this.reset();
        StringBuilder builder = new StringBuilder();
        builder.append(str);

        return builder.reverse().toString();
    }

    public List.ListNode moveToNext() {
        this.iterator = this.iterator.getNext();
        return this.iterator;
    }

    public void reset() {
        this.iterator = this.root;
    }

    public int getLength() {
        return this.length;
    }

    public List.ListNode getRoot() {
        return this.root;
    }

    public List.ListNode getIterator() {
        return this.iterator;
    }
}


public class Answer {
    public static List addTwoNumbers(List l1, List l2) {
        List result = null;

        if (l2.getLength() > l1.getLength()) {
            result = l1;
            l1 = l2;
            l2 = result;
            result = null;
        }

        boolean carry = false;
        l1.reset();
        l2.reset();

        while (l2.getIterator() != null) {
            int x = l2.getIterator().getVal() + l1.getIterator().getVal();
            if (carry) {
                x++;
            }
            
            if (x > 9) {
                x %= 10;
                carry = true;
            } else {
                carry = false;
            }

            if (result == null) {
                result = new List(x);
            } else {
                result.add(x);
            }

            l1.moveToNext();
            l2.moveToNext();
        }

        if (l1.getLength() != l2.getLength()) {
            int i = 0;
            while (i < l1.getLength() - l2.getLength()) {
                result.add(l1.getIterator().getVal());
				result.moveToNext();
				l1.moveToNext();
				i += 1;
            }
        }

        l1.reset();
        l2.reset();
        result.reset();
        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List list1 = null;
        List list2 = null;
        String inputString;

        System.out.print("Please enter the first number: ");
        inputString = Integer.toString(sc.nextInt());
        for (char c : inputString.toCharArray()) {
            if (list1 == null)
                list1 = new List(Integer.parseInt(String.valueOf(c)));
            else
                list1.addRoot(Integer.parseInt(String.valueOf(c)));
        }

        System.out.print("Please enter the second number: ");
        inputString = Integer.toString(sc.nextInt());
        for (char c : inputString.toCharArray()) {
            if (list2 == null)
                list2 = new List(Integer.parseInt(String.valueOf(c)));
            else
                list2.addRoot(Integer.parseInt(String.valueOf(c)));
        }

        System.out.println(list1.toString());
        System.out.println(list2.toString());

        List result = addTwoNumbers(list1, list2);
        System.out.println(result.toString());

        sc.close();
    }
}